app/assets/javascripts/discourse/models/topic-list.js.es6

```diff
@@ -174,6 +174,14 @@ TopicList.reopenClass({
         t.participants.forEach(p => (p.user = users[p.user_id]));
       }
 
+      // the same topic can show multiple times in the bookmark
+      // list because each post bookmark can have a different
+      // name + reminder setting, so we use the bookmark_id to make
+      // the topic more unique
+      if (result.filter === "bookmarks" && result.siteSettings.enable_bookmarks_with_reminders) {
```

Comment:
Paging @eviltrout, this is what I came up with in place of changing store, do you think this is a better place for this?

Reply:
I don't think this will work either, unfortunately. This code seems to be replacing the topic id with the bookmark id. It's still using the same identity map, so now another piece of code that uses the same id as the bookmark id is getting back something strange.

I think a better solution here would be to make a new route to list these bookmarks that doesn't use topic list. It would return a list of bookmarks, with each bookmark having a topic id that side loads to a topic object.